xarray/core/arithmetic.py

```diff
@@ -38,7 +38,8 @@ class SupportsArithmetic:
         numbers.Number,
         bytes,
         str,
-    ) + dask_array_type
+        dask_array_type,
+    )
```

Comment:
Explicit checking then raising of `NotImplementedError` is still require here else this happens

```
Error: test_xarray_handles_dask

TypeError: operand type(s) all returned NotImplemented from __array_ufunc__(<ufunc 'add'>, '__call__', <xarray.DataArray (x: 2, y: 2)>
array([[1., 1.],
       [1., 1.]])
Dimensions without coordinates: x, y, dask.array<ones_like, shape=(2, 2), dtype=float64, chunksize=(2, 2), chunktype=numpy.ndarray>): 'DataArray', 'Array'

Error: test_dask_defers_to_xarray

TypeError: operand type(s) all returned NotImplemented from __array_ufunc__(<ufunc 'add'>, '__call__', dask.array<ones_like, shape=(2, 2), dtype=float64, chunksize=(2, 2), chunktype=numpy.ndarray>, <xarray.DataArray (x: 2, y: 2)>
array([[1., 1.],
       [1., 1.]])
Dimensions without coordinates: x, y): 'Array', 'DataArray'
```

Reply:
Shouldn't this be all duck array types that support array_ufunc then?

Reply:
What do you mean exactly? The only way a type could pass `utils.is_duck_array` and not also support `__array_ufunc__` is that it has `ndim`, `shape`, `dtype` and either `__array_namespace__` or `__array_function__`.

Reply:
I think 
```
if not isinstance(x, self._HANDLED_TYPES + (SupportsArithmetic,)):
```
could be 
```
if not is_duck_array(x) and not isinstance(x, self._HANDLED_TYPES + (SupportsArithmetic,)):
```

that way we don't have to add array types to `_HANDLED_TYPES`. I also don't understand why dask is added but not sparse or cupy.